anvill/src/Optimize.cpp

```diff
@@ -84,63 +97,76 @@ void OptimizeModule(const EntityLifter &lifter_context,
     memory_escape->eraseFromParent();
   }
 
-  llvm::legacy::PassManager mpm;
-  mpm.add(llvm::createFunctionInliningPass(250));
-  mpm.add(llvm::createGlobalOptimizerPass());
-  mpm.add(llvm::createGlobalDCEPass());
-  mpm.add(llvm::createStripDeadDebugInfoPass());
-  mpm.run(module);
-
-  llvm::legacy::FunctionPassManager fpm(&module);
-  fpm.add(llvm::createDeadCodeEliminationPass());
-  fpm.add(llvm::createSinkingPass());
-  fpm.add(llvm::createNewGVNPass());
-  fpm.add(llvm::createSCCPPass());
-  fpm.add(llvm::createDeadStoreEliminationPass());
-  fpm.add(llvm::createSROAPass());
-  fpm.add(llvm::createEarlyCSEPass(true));
-  fpm.add(llvm::createBitTrackingDCEPass());
-  fpm.add(llvm::createCFGSimplificationPass());
-  fpm.add(llvm::createSinkingPass());
-  fpm.add(llvm::createCFGSimplificationPass());
-  fpm.add(llvm::createInstructionCombiningPass());
+  llvm::PassBuilder pass_builder;
+  llvm::LoopAnalysisManager lam;
+  llvm::FunctionAnalysisManager fam;
+  llvm::CGSCCAnalysisManager cgam;
+  llvm::ModuleAnalysisManager mam;
+  pass_builder.registerLoopAnalyses(lam);
+  pass_builder.registerFunctionAnalyses(fam);
+  pass_builder.registerCGSCCAnalyses(cgam);
+  pass_builder.registerModuleAnalyses(mam);
+  pass_builder.crossRegisterProxies(lam, fam, cgam, mam);
+
+  llvm::ModulePassManager mpm;
+  llvm::InlineParams inline_params;
+  inline_params.DefaultThreshold = 250;
+  mpm.addPass(llvm::ModuleInlinerWrapperPass(inline_params));
+  mpm.addPass(llvm::GlobalOptPass());
+  mpm.addPass(llvm::GlobalDCEPass());
+  // TODO(alex): Figure out what to do here.
+  // The new version of this pass only seems to have been added in LLVM 12.
+  // mpm.addPass(llvm::StripDeadDebugInfoPass());
```

Comment:
This pass was introduced in [this commit](https://github.com/llvm/llvm-project/commit/ba12e77ec16b38a4498610c6b8cdeb1a7e8a6aae). It'd be pretty easy to just write the wrapper ourselves if we wanted.